nautobot/extras/api/views.py

```diff
@@ -425,16 +425,25 @@ def comment(self, request, pk=None):
         """Add a comment to the specific stage (without approving or denying)."""
         stage = self.get_object()
 
+        if not stage.is_not_done_stage:
+            return Response(
+                {"detail": f"This stage is in {stage.state} state. Can't comment approved or denied stage."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
         comment = request.data.get("comments", "")
         if not comment:
             return Response(
                 {"detail": "Comment cannot be empty."},
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
-        ApprovalWorkflowStageResponse.objects.create(
-            approval_workflow_stage=stage, user=request.user, state=stage.state, comments=comment
+        approval_workflow_stage_response, _ = ApprovalWorkflowStageResponse.objects.get_or_create(
+            approval_workflow_stage=stage, user=request.user
         )
+        approval_workflow_stage_response.state = ApprovalWorkflowStateChoices.COMMENT
```

Comment:
This could potentially change an existing `APPROVED` response to `COMMENT` if the user is one of several required approvers. Should we maybe handle that case specially?